packages/grid/_modules_/grid/hooks/root/useGridApiEventHandler.ts

```diff
@@ -12,7 +13,15 @@ export function useGridApiEventHandler(
 
   React.useEffect(() => {
     if (handler && eventName) {
-      return apiRef.current.subscribeEvent(eventName, handler, options);
+      const enhancedHandler = (
```

Comment:
that could go out of this scope in a separate function. Not sure about redefining the handler as it changes the ref of the handler func potentially breaking immutability `if (handler === otherHandler)`  🤔

Reply:
We don't have any place where we compare the handler that is actually passed to `subscribeEvent`. I mean, compare the handler with the one that is saved in the event emitter. The ref would not change.

Another approach would be to move the `event.defaultMuiPrevented` validation to the event emitter, which is a very low level way to do it.